File: 0-warmup/2-pangram.cpp
// 09/22/24

#include<iostream>
#include<cctype>
#include<cstring>
using namespace std;

class Solution {
  public:
    bool checkIfPangram(string sentence) {
      // TODO: Write your code here
      vector<int>counter(25, 0);
      for (int i = 0; i < sentence.length(); i++) {
        if (sentence[i] == ' ') continue;
        int c = tolower(sentence[i]) - 'a';
        counter[c]++;
      }

      for (auto c : counter) {
        if (c < 1) return false;
      }
      return true;
    }
};

File: 0-warmup/2-pangram.py
# 09/22/24

class Solution:
  def checkIfPangram(self, sentence):
    # TODO: Write your code here
    '''letters = 'abcdefghijklmnopqrstuvwxyz'
    count = {}
    for c in letters:
      count[c] = 0

    for i in range(len(sentence)):
      if sentence[i].isalpha():
        count[sentence[i].lower()] += 1

    for c in letters:
      if count[c] < 1:
        return False

    return True'''

    seen = set()

    for c in sentence.lower():
      if c.isalpha():
        seen.add(c)

    if len(seen) == 26:
      return True

    return False
